services/cd-service/pkg/repository/transformer.go

```diff
@@ -1216,10 +1217,10 @@ func findOldApplicationVersions(state *State, name string) ([]uint64, error) {
 		return versions[i] >= oldestDeployedVersion
 	})
 
-	if positionOfOldestVersion < (keptVersionsOnCleanup - 1) {
+	if positionOfOldestVersion < (int(state.ReleaseVersionsLimit) - 1) {
 		return nil, nil
 	}
-	return versions[0 : positionOfOldestVersion-(keptVersionsOnCleanup-1)], err
+	return versions[0 : positionOfOldestVersion-(int(state.ReleaseVersionsLimit)-1)], err
```

Comment:
Let's add some unit tests for this? There are unit tests which check for the default limit of 20, but not with a custom value for `.ReleaseVersionsLimit`

Reply:
agree, there should be at least one test that calls the transformer and checks that files or releases were deleted

Reply:
I modified some tests to test for different values of kept versions.